### ISSUE TITLE
unexpected token '<' at line 1, col 1. (errors) [css/csslint]

### ISSUE THREAD

**Issue by PegasusWang** [posva/vim-vue#59]
```
<template>
  <div id="app">
    <img src="./assets/logo.png">
    <router-view></router-view>
  </div>
</template>

<script>
export default {
  name: 'app'
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>

```

hi，I use Syntastic with csslint and htmltidy, but when I open a vue file, I get this error, anyone can help, thks.

**Comment by adriaanzon** [comment]
I don't think this is possible with csslint

**Comment by zeayes** [comment]
`au BufNewFile,BufRead *.vue setf vue.html.javascript.css` this code causes.

**Comment by adriaanzon** [comment]
Ah, that's right.

Then this is another example of some of the side-effects of setting multiple filetypes.

What we're doing with multiple filetypes, is telling other plugins that the whole buffer may be treated as HTML and CSS and JavaScript at the same time. For some programs this is fine, for example:
- [ALE](https://github.com/w0rp/ale) with eslint and eslint-plugin-vue started working
- [tcomment](https://github.com/tomtom/tcomment_vim) started working correctly on all filetype regions

However, there's also some drawbacks:
- [Unexpected behavior when using ALE with proselint](https://github.com/w0rp/ale/issues/396)
- [Breaks autocmds](https://github.com/posva/vim-vue/pull/42#issuecomment-274355564)

So, setting multiple filetypes has its pros and cons. But I think it's trivial to offer some configuration in the `ftplugin` file, so that we can still use the desired features without having to use the multiple filetypes. 

For things like snippets, I think you should just add a vue.snippets with the following content: `extends html, css, javascript`. And if you're not writing your own snippets but use something like [honza/vim-snippets](https://github.com/honza/vim-snippets), maybe we could add such a file to that plugin.

Any thoughts @breneser? Do you think adding `ftplugin` configuration would be a good alternative to setting the filetype to `vue.html.javascript.css`?

**Comment by posva** [comment]
At the very least, we can also set the type to vue and html only and add a note on the readme about how to support snippets and others

**Comment by breneser** [comment]
@adriaanzon Sorry for late reply,

Excuse my ignorance on ftplugin configuration; I only started using Vim a while back and not quite sure what you refer to when you say `ftplugin` configuration.

But yes, setting filetype to vue sounds cleaner and feels more natural. 

And yes I am using honza/vim-snippets, if I can get snippets working with a couple lines of configuration, that is absolutely fine by me (a couple lines in readme explaining how to do that would be excellent for others like me). Or better yet having vue snippets in vim-snippets repo would a more out of the box experience which I initially thought this plugin would have when I made that PR.

**Comment by adriaanzon** [comment]
`ftplugin` is the directory for plugins that are local to a filetype, so it's useful for filetype-specific configuration. See `:h ftplugin` for more details.

Alright, just wanted to be sure I wasn't missing anything that's harder to configure without setting multiple filetypes. But now I realize that if there was, people could just set the multiple filetype manually 😄

I'll change it back and also add a note to the readme about how to set it to `vue.html.javascript.css` manually.

**Comment by adriaanzon** [comment]
Fixed in 32bc65cac29e16e2c9e4f91670578b19c5cc0aa4, e0c07d80c66b2f5e647ae32fd844e855ea709678
